### PR TITLE
docs(README): Improving readme with some demos and more info

### PR DESCRIPTION
**Pull request by m-strzelczyk**
Related to a request sent by one of our users. They note that GO library has a much better README with some actual info on how to use the library. That's why I decided to improve ours a little too.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -69,23 +69,112 @@ Windows
...
+an ``ExtendedOperation`` object that is returned when a long running operation is started.
+
+You can wait for the completion of an operation using ``result()`` method. This method accepts
+a ``timeout`` argument, specifying how long you want your process to wait for completion of the
+operation (in seconds). When the call to ``result()`` times out, the operation is not automatically
+cancelled. To cancel an operation you can use the ``cancel()`` method. You can query the operation
+at any time to check if it's complete with ``done()``.
+
+A sample method to handle LROs featuring error and warning reporting can be found in the Python
+code samples repository: [add link here]
```
Comment (rsamborski) [inline README.rst:147]: Link missing

File: README.rst [hunk 2]
```diff
@@ -69,23 +69,112 @@ Windows
...
+
+
+Code Samples
+~~~~~~~~~~~~
+You can find useful code samples that will demonstrate the usage of this library in `the
```
Comment (rsamborski) [inline README.rst:152]:
```suggestion
You can find useful code samples that will demonstrate the usage of this library on `the
```

File: README.rst [hunk 3]
```diff
@@ -69,23 +69,112 @@ Windows
...
+code sample browser`_.
```
Comment (rsamborski) [inline README.rst:153]:
```suggestion
Google Cloud samples page`_.
```

File: README.rst [hunk 4]
```diff
@@ -69,23 +69,112 @@ Windows
...
+
+.. _the code sample browser: https://cloud.google.com/docs/samples?language=python&product=computeengine
```
Comment (rsamborski) [inline README.rst:155]:
```suggestion
.. _the Google Cloud samples page: https://cloud.google.com/docs/samples?language=python&product=computeengine
```

I don't think we use Sample Browser name externally.

File: README.rst [hunk 5]
```diff
@@ -69,23 +69,112 @@ Windows
...
+
 PyCharm/JetBrains IDEs
 ~~~~~~~~~~~~~~~~~~~~~~
-Since the library has grown in size, the files it consists of have outgrown the [default size limit of ~2.5Mb](https://www.jetbrains.com/help/pycharm/file-idea-properties.html).
+Since the library has grown in size, the files it consists of have outgrown the `default size limit of ~2.5Mb`_.
 As a result, the code completion in JetBrains products can fail to work with the classes from our library. To
-fix this, you need to update the `idea.max.intellisense.filesize` setting in custom properties
-(Help -> Edit custom properties...). Just add a line like `idea.max.intellisense.filesize = 10000` to change this
+fix this, you need to update the ``idea.max.intellisense.filesize`` setting in custom properties
+(Help -> Edit custom properties...). Just add a line like ``idea.max.intellisense.filesize = 10000`` to change this
```
Comment (rsamborski) [inline README.rst:164]:
```suggestion
(Help -> Edit custom properties...). Just add the following line ``idea.max.intellisense.filesize = 10000`` to change this
```

**Review by rsamborski — APPROVED** [review]
Couple nits from my side. Otherwise LGTM.

File: README.rst (hunk 2 above)
Comment (savijatv) [reply README.rst:152]: You can find useful code samples that demonstrate the usage of this...

File: README.rst [hunk 6]
```diff
@@ -69,23 +69,112 @@ Windows
...
 
 -  Read the `Client Library Documentation`_ for Compute Engine API
-   API to see other available methods on the client.
+   to see other available methods on the client.
 -  Read the `Compute Engine API Product documentation`_ to learn
    more about the product and see How-to Guides.
 -  View this `README`_ to see the full list of Cloud
    APIs that we cover.
 
 .. _Compute Engine API Product documentation:  https://cloud.google.com/compute/
```
Comment (savijatv) [inline README.rst:179]: Or can we give this link - https://cloud.google.com/compute/docs/api/libraries

File: README.rst (hunk 1 above)
Comment (m-strzelczyk) [reply README.rst:147]: Waiting for https://github.com/GoogleCloudPlatform/python-docs-samples/pull/9104

**Review by kweinmeister — APPROVED** [review]

File: README.rst [hunk 7]
```diff
@@ -69,23 +69,117 @@ Windows
...
 
 
+Authentication and Authorization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This client library supports authentication via `Google Application Default Credentials`_
+, or by providing a JSON
+key file for a Service Account. See examples below.
+
+Google Application Default Credentials (ADC) is the recommended way to authorize and authenticate
+clients. To learn how to create and obtain `Google Application Default Credentials`_. Here is an
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply README.rst:79]: Agreed. Applied.

File: README.rst (hunk 1 above)
Comment (m-strzelczyk) [reply README.rst:152]: Link added.

File: README.rst [hunk 8]
```diff
@@ -139,17 +137,19 @@ see `Scopes documentation page`_. Example:
...
-an ``ExtendedOperation`` object that is returned when a long running operation is started.
+Long-Running Operations (LROs), like the many ``insert()`` operations, can be handled using
+the ``ExtendedOperation`` object that is returned when the LRO is started.
 
-You can wait for the completion of an operation using ``result()`` method. This method accepts
+You can wait for the completion of an operation using its ``result()`` method. This method accepts
 a ``timeout`` argument, specifying how long you want your process to wait for completion of the
 operation (in seconds). When the call to ``result()`` times out, the operation is not automatically
-cancelled. To cancel an operation you can use the ``cancel()`` method. You can query the operation
+cancelled. At any time, you can check whether the operation is complete by using its ``done()`` method.
```
Comment (vchudnov-g) [inline README.rst:146]: There's a sentence fragment cruft at the end of the paragraph.

**Review by vchudnov-g — APPROVED** [review]
LGTM, but please do remove a cruft sentence fragment left over from a previous revision (see comment)

**Pull request merged**